CMakeLists.txt

```diff
@@ -74,36 +75,11 @@ if(NOT THEROCK_USE_ROCM_LIBRARIES)
   )
 endif()
 
-# Use the rccl amd rccl-test repos tracked in TheRock's `.gitmodules` or
-# specify alternative source locations.
-option(THEROCK_USE_EXTERNAL_RCCL "Use external rccl source locations" OFF)
-set(THEROCK_RCCL_SOURCE_DIR_DEFAULT "${THEROCK_SOURCE_DIR}/comm-libs/rccl")
-set(THEROCK_RCCL_TESTS_SOURCE_DIR_DEFAULT "${THEROCK_SOURCE_DIR}/comm-libs/rccl-tests")
-
-if(THEROCK_USE_EXTERNAL_RCCL)
-  if(NOT THEROCK_RCCL_SOURCE_DIR)
-    message(FATAL_ERROR "If THEROCK_USE_EXTERNAL_RCCL is set, THEROCK_RCCL_SOURCE_DIR is required!")
-  endif()
-  if(NOT THEROCK_RCCL_TESTS_SOURCE_DIR)
-    message(FATAL_ERROR "If THEROCK_USE_EXTERNAL_RCCL is set, THEROCK_RCCL_TESTS_SOURCE_DIR is required!")
-  endif()
-endif()
-
-set(THEROCK_RCCL_SOURCE_DIR "${THEROCK_RCCL_SOURCE_DIR_DEFAULT}" CACHE STRING "Path to rccl sources")
-set(THEROCK_RCCL_TESTS_SOURCE_DIR "${THEROCK_RCCL_TESTS_SOURCE_DIR_DEFAULT}" CACHE STRING "Path to rccl-tests sources")
-cmake_path(ABSOLUTE_PATH THEROCK_RCCL_SOURCE_DIR NORMALIZE)
-cmake_path(ABSOLUTE_PATH THEROCK_RCCL_TESTS_SOURCE_DIR NORMALIZE)
-
-# Use the composable_kernel tracked in TheRock's `.gitmodules` or
-# specify an alternative source location.
-option(THEROCK_USE_EXTERNAL_CK "Use external CK source location" OFF)
-set(THEROCK_CK_SOURCE_DIR_DEFAULT "${THEROCK_SOURCE_DIR}/ml-libs/composable_kernel")
-
-if(THEROCK_USE_EXTERNAL_CK AND NOT THEROCK_CK_SOURCE_DIR)
-  message(FATAL_ERROR "If THEROCK_USE_EXTERNAL_CK is set, THEROCK_CK_SOURCE_DIR is required!")
-endif()
-set(THEROCK_CK_SOURCE_DIR "${THEROCK_CK_SOURCE_DIR_DEFAULT}" CACHE STRING "Path to ck sources")
-cmake_path(ABSOLUTE_PATH THEROCK_CK_SOURCE_DIR NORMALIZE)
+# Allow to specify alternative source locations instead of using
+# repositories tracked in TheRock's `.gitmodules`.
+therock_enable_external_source("rccl" "${THEROCK_SOURCE_DIR}/comm-libs/rccl" OFF)
```

Comment:
The new generated `THEROCK_{PROJECT}_SOURCE_DIR` options could be documented either at https://github.com/ROCm/TheRock/blob/main/docs/development/build_system.md or https://github.com/ROCm/TheRock/blob/main/README.md#optional-configuration-flags